scripts/engine/utility.py

```diff
@@ -144,45 +145,96 @@ def get_chebyshev_distance(start_pos: Tuple[int, int], target_pos: Tuple[int, in
     return scipy.spatial.distance.chebyshev(start_pos, target_pos)
 
 
-def get_coords_from_shape(shape: ShapeType, size: int) -> List[Tuple[int, int]]:
+def _calculate_square_shape(size: int) -> List[Tuple[int, int]]:
     """
-    Get a list of coords from a shape and size.
+    Calculate all the tiles in the range of a square
     """
-    list_of_coords = []
+    coord_list = []
+    width = size
+    height = size
 
-    if shape == Shape.TARGET:
-        list_of_coords.append((0, 0))  # single target, centred on selection
+    for x in range(-width, width + 1):
+        for y in range(-height, height + 1):
+            coord_list.append((x, y))
+    return coord_list
 
-    elif shape == Shape.SQUARE:
-        width = size
-        height = size
 
-        for x in range(-width, width + 1):
-            for y in range(-height, height + 1):
-                list_of_coords.append((x, y))
+def _calculate_circle_shape(size: int) -> List[Tuple[int, int]]:
+    """
+    Calculate all the tiles in the range of a circle
+    """
+    coord_list = []
+    radius = (size + size + 1) / 2
+
+    for x in range(-size, size + 1):
+        for y in range(-size, size + 1):
+            if x * x + y * y < radius * radius:
+                coord_list.append((x, y))
 
-    elif shape == Shape.CIRCLE:
-        radius = (size + size + 1) / 2
+    return coord_list
 
-        for x in range(-size, size + 1):
-            for y in range(-size, size + 1):
-                if x * x + y * y < radius * radius:
-                    list_of_coords.append((x, y))
 
-    elif shape == Shape.CROSS:
-        x_coords = [-1, 1]
+def _calculate_cross_shape(size: int) -> List[Tuple[int, int]]:
+    """
+    Calculate all the tiles in the range of a cross
+    """
+    coord_list = [(0, 0)]
+    x_coords = [-1, 1]
+
+    for x in x_coords:
+        for y in range(-size, size + 1):
+
+            # ignore 0's to ensure no duplication when running through the range
+            # the multiplication of x by y means they are always both 0 if y is
+            if y != 0:
+                coord_list.append((x * y, y))
+
+    return coord_list
+
+
+def _calculate_cone_shape(size: int, direction: Tuple[int, int]) -> List[Tuple[int, int]]:
```

Comment:
Since cones are not symmetric we needed the direction of the cone in other to calculate the shape.

Here is a simple explanation of how this code works:
We start with just one tile at (0, 0).
Each iteration from [1, size] we:
- Take the last row of the cone and add all the tiles that are 1 tile away in the provided direction
- Foreach tile of the tiles we just added we add the tile perpendicular to it into the list (we could replace this by just adding the perpendiculars of the first and last tiles from the row.)
- We push this list into the cone

The magic of this code is that it works in all the directions. I can make a visual example later if you want to see it.